src/mango/src/mango_idx.erl

```diff
@@ -68,7 +68,7 @@ get_usable_indexes(Db, Selector, Opts) ->
     UsableFilter = fun(I) -> is_usable(I, Selector, SortFields) end,
     UsableIndexes1 = lists:filter(UsableFilter, UsableIndexes0),
 
-    case maybe_filter_by_sort_fields(UsableIndexes1, SortFields) of
+    case maybe_filter_by_sort_fields(UsableIndexes1, Selector, SortFields) of
```

Comment:
My bad that I was vague on this earlier. What i was suggesting here was to move maybe_filter_by_sort_fields completely and just move that logic to the existing `is_usable` callback that already has all the information required. The thought was that `is_usable` shouldn't return true if the index can't satisfy the sort (cause then its not usable :).